wolfcrypt/src/port/st/STM32MP13.md

```diff
@@ -0,0 +1,212 @@
+# STM32MP13 Port
+
+The STM32MP13 is unique in that it is an MPU instead of an MCU. The HAL also
+behaves a little differently. This document outlines how to use it in bare
+metal mode. For Linux, this should be used as a normal ARM Linux device.
+
+## Linux
+
+To cross-compile from a Linux host to the STM32MP13 OpenSTLinux, you need to
+install the [SDK](https://www.st.com/en/embedded-software/stm32mp1dev.html#get-software).
+In this example, I have extracted it to `/opt/st`.
+
+Your build environment is configured by running:
+
+```sh
+source /opt/st/stm32mp1/4.2.4-openstlinux-6.1-yocto-mickledore-mpu-v24.06.26/environment-setup-cortexa7t2hf-neon-vfpv4-ostl-linux-gnueabi
+```
+
+If you wish to compile with support for `/dev/crypto` then you will also need to
+do the following so that the headers are found by the compiler:
+
+```sh
+export CFLAGS="$CFLAGS -I /opt/st/stm32mp1/4.2.4-openstlinux-6.1-yocto-mickledore-mpu-v24.06.26/sysroots/cortexa7t2hf-neon-vfpv4-ostl-linux-gnueabi/usr/src/debug/cryptodev-module/1.12-r0/"
+```
+
+When running `./configure`, make sure you add `--host=arm-linux-gnueabi` to the
+configure options.
+
+## Bare metal
+
+### Setting up
+
+The board itself has dip switches to set the boot mode. These should be set to
+off-off-on-off to set the board into "engineering mode". The MPU's SRAM can
+then be flashed via the ST-Link.
+
+#### Device Configuration Tool
+
+In the configuration tool, enable and activate the following:
+
+```
+CRYP1
+HASH1
+PKA
+RNG1
+```
+
+#### DDR RAM
```

Comment:
Instead of forcing users to start from the DDR_Init example please setup a new example at http://github.com/wolfSSL/wolfssl-examples-stm32

Reply:
I'll rewrite to make this to make it clearer, but the DDR_Init project is not a starting point. It is a separate project you need to run first as-is before running your wolfSSL project, every time you power the board on. There is no way around this for engineering mode.

The DDR_Init project sets up the clocks and initializes the DDR RAM only. Then a blue LED flashes to say it is done. When you run the wolfSSL project, the ST-Link loads this directly into the DDR RAM and runs it from there. This is why the DDR_Init project is required to run first, otherwise the board will immediately crash :)